Advanced-Recruit-Training/src/LogFileReader.cpp

```diff
@@ -62,10 +62,28 @@ bool LogFileReader::parseLine(const QString& line, BatteryData& batteryData) con
     QString timeString = sections.at(0);
     batteryData.time = QTime::fromString(timeString, STRING_TIME_FORMAT);
 
-    batteryData.voltage = sections.at(1).toDouble();
+    if(sections.length() < COLUMNS || sections.length() > COLUMNS)
+        return false;
 
-    batteryData.current = sections.at(2).toDouble();
+    bool DoubleConversion1;
+    bool DoubleConversion2;
 
-    return true;
+    batteryData.voltage = sections.at(1).toDouble(&DoubleConversion1);
+
+    batteryData.current = sections.at(2).toDouble(&DoubleConversion2);
+
+    if(DoubleConversion1 != true || DoubleConversion2 != true)
+        return false;
```

Comment:
Put curly brackets for all if statements even if there is only one line of code in the statement